CMakeLists.txt

```diff
@@ -357,8 +357,8 @@ if( BUILD_RUBY )
     # brew install autoconf automake libtool
     # brew link autoconf automake libtool
     ExternalProject_Add(Ruby
-      URL https://codeload.github.com/ruby/ruby/tar.gz/v2_5_1
-      URL_MD5 98b550a05fec0f7a43a22d6c37404fd4
+      URL https://codeload.github.com/ruby/ruby/tar.gz/v2_5_5
+      URL_HASH SHA256=A3BA7160C7E0CE03865CDABA9B7C26974A8A0FC23FC953269E0463C5ACAFB7A1
```

Comment:
@lefticus can't we keep our md5s?

Reply:
I didn't have an MD5 checksum tool installed on Windows, and MD5 is pretty well known to be broken at this point, so I just used SHA256. I can change it back if it matters tho.

Reply:
Just noticed that SHA256 is also consistent with the other dependencies in this file.

Reply:
@macumber personally I would stick with sha256 for the reasons mentioned above, for the things that *needed* to get changed. To minimize changes and risk I did revert to 2.5.1 on Unix.